### ISSUE TITLE
Customize link URL for different trips

### ISSUE THREAD

**Issue by markstm17** [ArgoBus/Argo-Project-V0#236]
## This is what it currently looks like

![better link url](https://cloud.githubusercontent.com/assets/11848699/10647497/55cbc37c-77fe-11e5-9e00-aab05d05a10f.png)

https://argobus.com/trips/pledge/18
--

#### We should link it to whatever the organizer chooses the trip name to be _(Personalization and customization component behind it)_ on Create trip page

![screen shot 2015-10-21 at 2 18 09 pm](https://cloud.githubusercontent.com/assets/11848699/10647583/be245ea2-77fe-11e5-986d-509a0dc55a0b.png)

# Examples

**Comment by duncangraham** [comment]
The issue with your implementation is that if there are 2 trips with the same name (which isn't going to be uncommon) the URLs cease to be unique.

I agree that the structure Blade made isn't ideal, but I'm not sold that concatenated strings is the best execution. Looking at airbnb and facebook for implementations, both use unique identifiers

https://www.facebook.com/groups/604798506278511/?ref=br_rs
https://www.airbnb.com/rooms/8581302

the way they should probably be ordered is:

argobus.com/trips/1092273

The URL won't need to communicate the event most likely, because people will share on social media like: 'Lets party our way to the packers game this weekend! argobus.com/trips/1092273'

**Comment by duncangraham** [comment]
the URL format is now www.argobus.com/trips/:trip_id

we can look into implementing vanity URLS at a later time. (potentially for a surcharge)

**Comment by markstm17** [comment]
Is :trip_id simply a number then?

Jake's this past weekend was trips/8

Just curious @duncangraham

**Comment by duncangraham** [comment]
yea, it's the unique identifier

**Comment by markstm17** [comment]
When we have pick up locations set up through our system, we can have customized URL's.

Just bringing this up to our/your attention again because when there are multiple trips from a singular bar/pick up location when friends get a text, an email or see a post about a specific trip, it is just one more communication point for us to communicate with our users/pledges.

I envision it like this:

www.argobus.com/trips/kollege-klub-madison-brewers-tailgate-april-4th

The trips organically become customized through the name of the bar, name of city bar is in, name of the event we call it (Brewers Tailgate) and the Date (All through Argo through our system)

Not essential right now, but reopneing because i think in the next 4-6 months it should change from 

trips/8

To

What I have stated above

**Comment by duncangraham** [comment]
Yea I agree it would be nice, but I need to figure out a way that these names would continue to be unique. Your example doesn't work if there are two buses at that location on that day. There will likely be other edge cases that could break things as well. 

This isn't a trivial problem. It'll need a lot of thought and planning. I think implementation will likely also require a decent amount of sweat. I'll keep the ticket open, but I'm punting on this

**Comment by duncangraham** [comment]
Just a thought-- rdio handles this problem by adding the id and the name like so

http://www.rdio.com/people/DuncanGraham90/playlists/15154789/november_2015/

I think I might be able to append the location and date on the end of the URL 

argobus.com/trips/8/kollege-klub-madison-brewers-tailgate-april-4th

**Comment by markstm17** [comment]
Yeah I agree for now @duncangraham , definitely don't worry about it-punt for now because of higher priority tickets! 

I just wanted to readdress the ticket and reopen it because it needs to be something we fix in the relatively near future. 

And for

> Just a thought-- rdio handles this problem by adding the id and the name like so

> http://www.rdio.com/people/DuncanGraham90/playlists/15154789/november_2015/

> I think I might be able to append the location and date on the end of the URL 

> argobus.com/trips/8/kollege-klub-madison-brewers-tailgate-april-4th

# I think that is a step in the right direction

**Comment by markstm17** [comment]
Pick up location and name of the bar for trips at those bars NEVER changes.
- What changes is:
 - The event (Brewers game or Packers Game or Jimmy Buffett Concert)
 - The Date
